### PR TITLE
Lua modf

### PR DESCRIPTION
**Pull request by SusheelThapa**
### Description

- Closed issue #3500 
- Created documentation for the `math.modf()` function in Lua.
- Included details on usage, syntax, and examples.
- Motivation: To assist users in understanding the functionality of `math.modf()` in Lua.
- Context: Aligning the documentation with existing math function documentation in Lua.

### Type of Change

- Adding a new entry

### Checklist

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/lua/concepts/mathematical-library/terms/modf/modf.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+---
+Title: 'modf()'
+Description: 'Used to extract the integral and fractional parts of a given number'
```
Comment (caupolicandiaz) [inline content/lua/concepts/mathematical-library/terms/modf/modf.md:3]:
```suggestion
Description: 'Used to extract the integral and fractional parts of a given number.'
```

File: content/lua/concepts/mathematical-library/terms/modf/modf.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  - 'Arithmetic'
+  - 'Functions'
+  - 'Math'
+  - 'Numbers'
+CatalogContent:
+  - 'learn-lua'
+  - 'paths/computer-science'
+---
+
+The **`math.modf()`** function is a part of the Lua `math` library and is used to split a given number into **its integral and fractional parts**. It rounds the number down to the nearest integer and returns both the integer part and the fractional part as separate values.
```
Comment (caupolicandiaz) [inline content/lua/concepts/mathematical-library/terms/modf/modf.md:17]:
```suggestion
The **`math.modf()`** function is a part of the Lua `math` library and is used to split a given number into *its integral and fractional parts*. It rounds the number down to the nearest integer and returns both the integer part and the fractional part as separate values.
```

File: content/lua/concepts/mathematical-library/terms/modf/modf.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+> **Note:** `math.modf()`rounds the number down and always returns a value equal to or less than the given number. If you need to round numbers in either direction, consider using `math.round()`.
```
Comment (caupolicandiaz) [inline content/lua/concepts/mathematical-library/terms/modf/modf.md:19]:
```suggestion
> **Note:** `math.modf()` rounds the number down and always returns a value equal to or less than the given number. 
```

Comment (caupolicandiaz) [reply content/lua/concepts/mathematical-library/terms/modf/modf.md:19]: I believe that math.round is only available in luau

File: content/lua/concepts/mathematical-library/terms/modf/modf.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+```
+
+## Example 1
+
+To find the integral and fractional parts of a number:
+
+```lua
+local num = 5.3
+local integerPart, fractionalPart = math.modf(num)
+print("Integer part: " .. integerPart)       -- Output: Integer part: 5
```
Comment (caupolicandiaz) [inline content/lua/concepts/mathematical-library/terms/modf/modf.md:36]: generally we use ```shell blocks for showing the output.

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@SusheelThapa I've completed my initial review. Please update the entry, and tag me when you've finished making changes thanks.

**Comment by SusheelThapa** [comment]
@caupolicandiaz Can you please review, I have updated as per your suggestion?

**Review by caupolicandiaz — APPROVED** [review]
@SusheelThapa This is ready for a second review.

File: content/lua/concepts/mathematical-library/terms/modf/modf.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+> **Note:** `math.modf()` rounds the number down and always returns a value equal to or less than the given number.
+
+## Syntax
+
+As `modf()` is a method that is a part of the standard Lua `math` library, it must be called as `math.modf()`.
+
+```pseudo
+integerPart, fractionalPart = math.modf(number) -- Returns the intergal and fractional part
+```
+
```
Comment (SSwiniarski) [inline content/lua/concepts/mathematical-library/terms/modf/modf.md:28]:
````suggestion
integerPart, fractionalPart = math.modf(number)
```

`math.modf()` returns the integral part of `number` in `integerPart` and the fractional part in `fractionalPart`.

````

Prefer the explanatory text outside the code block.

File: content/lua/concepts/mathematical-library/terms/modf/modf.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+This results in the following output:
+
+```shell
+Output: Integer part: 8
+Output: Fractional part: 0.0
+```
+
+## Example 3
+
+For negative numbers, `math.modf()` returns the closest integer less than or equal to the given value:
```
Comment (SSwiniarski) [inline content/lua/concepts/mathematical-library/terms/modf/modf.md:67]:
```suggestion
For negative numbers, `math.modf()` returns the closest integer greater than or equal to the given value:
```
-3.0 is greater than -3.3

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@SusheelThapa I did a review and left a couple of comments.

**Comment by SusheelThapa** [comment]
@SSwiniarski I have updated based in you suggestions. Can you review it?

**Review by SSwiniarski — APPROVED** [review]
@SusheelThapa Looks good.

**Pull request merged**